tests/test_parser.py

```diff
@@ -131,6 +136,18 @@ def test_retains_casing(self):
             retain_casing=True
         )) == [('Three', ), ('letter', ), ('acronym', ), ('TLA',)]
 
+    def test_retains_punctuation(self):
+        assert list(textparser.word_tokenize(
+            text='who, where? (question!)',
+            retain_punctuation=True
+        )) == [('who', ), (',', ), ('where',), ('?', ), ('(',), ('question',), ('!',), (')',)]
+
+    def test_retains_punctuation_within_tokens(self):
+        assert list(textparser.word_tokenize(
+            text='is the oven pre-heated?',
+            retain_punctuation=True
+        )) == [('is', ), ('the', ), ('oven',), ('pre-heated', ), ('?',)]
```

Comment:
why do we dinstinguish between `-` and other punctuation here?

Reply:
The difference (and it's debatable whether this is the correct way to handle things) is that the `-` here is punctuation encountered inside a word token.  Currently that value (and also `\` and `/`) are handled as special cases (see https://github.com/MichaelAquilina/hashedindex/pull/16/files#diff-bca7df3b022757454ef625ad3579a5b5R27-L29)

Reply:
This should probably be documented somewhere (I cant remember my justification for having these exceptions as its been a long time!)